src/adagrams.js

```diff
@@ -1,15 +1,153 @@
+/*constructing a leeterpool object with letter as key and no. of times it can be drawn as value*/
+const LETTERPOOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const drawnLetters = [];
+  //Evaluating total size of probable letters can be drawn
+  let totalSize = Object.values(LETTERPOOL).reduce((acc, val) => acc + val, 0);
+  let letterPoolArr = Object.entries(LETTERPOOL);
+  let count = 0,
+    letterIdx;
+  const handSize = 10;
+
+  while (drawnLetters.length < handSize) {
+    if (totalSize === 0) {
+      console.log('No more letters available to draw.');
+      break;
+    }
+
+    letterIdx = Math.random() * totalSize;
+    for (const [letter, frequency] of letterPoolArr) {
+      count += frequency;
+      if (letterIdx < count) {
+        drawnLetters.push(letter);
+        totalSize -= 1;
+        //updating array with already drawn letters
+        const updatedArr = letterPoolArr.map(([l, freq]) =>
+          l === letter ? [l, freq - 1] : [l, freq]
+        );
+        letterPoolArr = updatedArr.filter(([l, freq]) => freq > 0);
+        break;
+      }
+    }
+  }
+  return drawnLetters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  for (let letter of lettersInHand) {
+    letter.toUpperCase();
+  }
+  const letterObj = {};
+  for (let letter of lettersInHand) {
+    letterObj[letter] ? letterObj[letter]++ : (letterObj[letter] = 1);
+  }
+ //Utilizing object to track the letters drawn
+  for (let letter of input) {
+    letter = letter.toUpperCase();
```

Comment:
Prefer to call the method on line 75.

```suggestion
  for (let letter of input.toUpperCase()) {
```